trestle/core/control_io.py

```diff
@@ -807,71 +737,236 @@ def _read_sections(ii: int, lines: List[str], control_id: str,
             control_parts = None
         return ii, control_parts
 
+    @staticmethod
+    def _clean_prose(prose: List[str]) -> List[str]:
+        new_prose = []
+        # remove all known bad lines
+        for line in prose:
+            if not line.startswith(const.SSP_ADD_IMPLEMENTATION_PREFIX) and not line.startswith('___'):
+                new_prose.append(line)
+        # remove empty lines at start and end
+        for _ii, line in enumerate(new_prose):
+            if line.strip():
+                break
+        prose = new_prose[_ii:]
+        for _ii, line in enumerate(reversed(prose)):
+            if line.strip():
+                break
+        clean_prose = prose[:len(prose) - _ii]
+        # if there is no useful prose this will return [''] and allow generation of a statement with empty prose
+        return clean_prose
+
```

Comment:
@enikonovad - i'd like you to look at this and see what should be moved to the markdown node api.

Reply:
It's somewhat specific to what's going on here - but if it can be generalized that's fine.  Sometimes you don't want to strip empty lines.

Reply:
Looking at this code I can see a certain benefit if we move this to the `MarkdownNode`. In my understanding this code simply removes empty lines and two specific lines formats. 
I'm thinking this logic can be moved to another block in `MarkdownNode` and a new `control specific text` variable in `SectionContent` as follows:

```
def _build_tree(
        self,
        lines: List[str],
        root_key: str,
        starting_line: int,
        level: int,
        governed_header: Optional[str] = None
    ) -> Tuple[MarkdownNode, int]:
...
...
...
          else:
                content.text.append(line)
                if line:
                    if not line.startswith(const.SSP_ADD_IMPLEMENTATION_PREFIX) and not line.startswith('___'):
                        content.control_text.append(line)
                 
                i += 1
```

Reply:
Then this text can be directly extracted from `node.content.control_text`

Reply:
Specifically the code removes empty lines at the start and at the end, but leaves interior empty lines alone.  This may have general need, so yes it could be in the markdown.
Separately it completely removes horizontal rule lines - which also may be a general need.
And it removes all lines containing a given string.
If these three things can be accommodated in the api that would be fine.  But of all these, removing lines containing a specific string may not be a general need.

Reply:
I see in such a case the logic is a bit different, we can still add it right before returning the node:
```
<here> content.clean_prose = clean_prose(content.text)
md_node = MarkdownNode(key=root_key, content=content)
md_node.subnodes = node_children
return (md_node, i)
```

However, it would depend on whether this logic will be used anywhere else and trade-off of using more RAM to store text is better than cleaning it in `control_io` afterwards. If the benefit is minimal, I think it is fine to keep it as is